### PR TITLE
Cleanup for Algorand branch (libcorev1)

### PR DESCRIPTION
**Pull request by haammar-ledger**
A bit of cleanup before opening the PR to **ledgerhq**.

Don't mind the massive lot of generated JNI files...

If you see any further cleanup opportunity please tell me.

### CODE REVIEW

File: core/src/wallet/algorand/AlgorandAccount.cpp [hunk 1]
```diff
@@ -386,7 +385,7 @@ namespace algorand {
 
         auto self = std::static_pointer_cast<Account>(shared_from_this());
         _synchronizer->synchronizeAccount(self)->getFuture()
```
Comment (barjonr) [inline core/src/wallet/algorand/AlgorandAccount.cpp:387]: You could use `getSelf()` directly. Also, `static_pointer_cast` is actually better than `dynamic_pointer_cast`, could you change it in `getSelf` please?

File: core/idl/wallet/algorand/idl.djinni [hunk 2]
```diff
@@ -4,11 +4,10 @@
 @import "configuration.djinni"
 
 AlgorandConfigurationDefaults = record {
-    # TODO Switch to mainnet at some point
-    const ALGORAND_API_ENDPOINT: string = "https://testnet-algorand.api.purestake.io/ps1/v1/";
+    const ALGORAND_API_ENDPOINT: string = "https://mainnet-algorand.api.purestake.io/ps1/v1/";
```
Comment (barjonr) [inline core/idl/wallet/algorand/idl.djinni:7]: I think this will break all our explorer tests, right?

Comment (haammar-ledger) [reply core/idl/wallet/algorand/idl.djinni:7]: Damn you're right, I didn't run the tests with this PR indeed. I'll work on that.

File: core/src/wallet/algorand/AlgorandAccount.cpp (hunk 1 above)
Comment (haammar-ledger) [reply core/src/wallet/algorand/AlgorandAccount.cpp:387]: :+1:

**Pull request merged**